engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OpenIdConnectService.scala

```diff
@@ -0,0 +1,33 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.Decoder
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.security.oauth2.BaseOAuth2Service.millisToDeadline
+
+import scala.concurrent.duration.Deadline
+import scala.concurrent.{ExecutionContext, Future}
+
+trait OpenIdConnectAccessTokenResponse extends StandardAccessTokenResponse {
+  val id_token: Option[String]
+}
+
+class OpenIdConnectService[
+  ProfileResponse: Decoder,
+  AccessTokenResponse <: OpenIdConnectAccessTokenResponse : Decoder,
+  DecodedJwtAccessToken <: JwtStandardClaims : Decoder
+](clientApi: OAuth2ClientApi[ProfileResponse, AccessTokenResponse],
+  createLoggedUser: (ProfileResponse) => LoggedUser,
+  configuration: OAuth2Configuration)
+ (implicit ec: ExecutionContext)
+  extends JwtOAuth2Service[ProfileResponse, AccessTokenResponse, DecodedJwtAccessToken](clientApi, createLoggedUser, configuration)
+    with LazyLogging {
+
+  protected val useIdToken: Boolean = configuration.jwt.exists(_.userinfoFromIdToken)
```

Comment:
private instead of protected?

Reply:
If we allow people to implement their own implementation of OAuth2 and inject it by SPI it desirable to give a chance not to write it all from scratch.